### DISCUSSION TITLE
Changing how Harpoon Gun work

### DISCUSSION THREAD

**Discussion by AFK-TankyMOFO** [FakeFishGames/Barotrauma#9069] [Ideas and suggestions]
As of now, harpoon gun fires harpoon with rope that will reel in the harpoon if it hits a creature, along with the creature it hits if you continue holding down the "Aim" button and haven't fired another harpoon yet.

But the problem is, this actually helps the creature you are shooting. Since by reeling them in, you are closing in the distance between you, someone with a harpoon gun, which is a ranged weapon, and a creature that most likely has metal rending-sharp beaks, or claws and is now extremely pissed now what with the harpoon lodged in their forehead.

Now, this problem can be remedied by letting go of the "Aim" button as you fire the harpoon as it cuts off the rope immediately, but at the same time, you probably want to keep your gun trained on your target.

My suggestion is making it so that holding the "Shoot" button reel in the rope, and letting go of the button cuts off the rope, since harpoon gun is not a automatic weapon and there is no benefit of holding down the "Shoot" button otherwise.

**Comment by ghost** [comment]
> harpoon gun is not a automatic weapon

Aren't all weapons automatic?

**Reply by AFK-TankyMOFO** [reply]
In a sense, but it only fires one shot every three seconds.

**Comment by Saevon** [comment]
Does this affect the harpoon coil rifle?  did those also pull you in?

**Comment by jigardi** [comment]
Hmmmm... So you basically can't use the weapon as you'd intuitively expect to?  Is that a good summary of the problem?

**Comment by girlyguppy** [comment]
but this would mean you could only reel them in until the next harpoon shoots. unless continuing to hold would result in more than 1 rope and harpoon being hooked at once, therefore allowing you to reel them faster. i support that, it would be cool to shoot an enemy 5 times and then drag them around
